lasagne/layers/conv.py

```diff
@@ -1047,88 +1352,47 @@ class TransposedConv3DLayer(BaseConvLayer):  # pragma: no cover
-                               stride, crop, untie_biases, W, b,
-                               nonlinearity, flip_filters, n=3, **kwargs)
-        # rename self.pad to self.crop:
-        self.crop = self.pad
-        del self.pad
-
-    def get_W_shape(self):
-        num_input_channels = self.input_shape[1]
-        # first two sizes are swapped compared to a forward convolution
-        return (num_input_channels, self.num_filters) + self.filter_size
-
-    def get_output_shape_for(self, input_shape):
-        if self.output_size is not None:
-            size = self.output_size
-            if isinstance(self.output_size, T.Variable):
-                size = (None, None, None)
-            return input_shape[0], self.num_filters, size[0], size[1], size[2]
-
-        # If self.output_size is not specified, return the smallest shape
-        # when called from the constructor, self.crop is still called self.pad:
-        crop = getattr(self, 'crop', getattr(self, 'pad', None))
-        crop = crop if isinstance(crop, tuple) else (crop,) * self.n
-        batchsize = input_shape[0]
-        return ((batchsize, self.num_filters) +
-                tuple(conv_input_length(input, filter, stride, p)
-                      for input, filter, stride, p
-                      in zip(input_shape[2:], self.filter_size,
-                             self.stride, crop)))
 
-    def convolve(self, input, **kwargs):
-        border_mode = 'half' if self.crop == 'same' else self.crop
-        op = T.nnet.abstract_conv.AbstractConv3d_gradInputs(
-            imshp=self.output_shape,
-            kshp=self.get_W_shape(),
-            subsample=self.stride, border_mode=border_mode,
-            filter_flip=not self.flip_filters)
-        output_size = self.output_shape[2:]
-        if isinstance(self.output_size, T.Variable):
-            output_size = self.output_size
-        elif any(s is None for s in output_size):
-            output_size = self.get_output_shape_for(input.shape)[2:]
-        conved = op(self.W, input, output_size)
-        return conved
+        super(TransposedConv3DLayer, self).__init__(
+                incoming, num_filters, filter_size, stride, crop, untie_biases,
+                W, b, nonlinearity, flip_filters, output_size=output_size,
+                n=3, **kwargs)
 
 Deconv3DLayer = TransposedConv3DLayer
 
 
 if not hasattr(T.nnet.abstract_conv,
+               'AbstractConv_gradInputs'):  # pragma: no cover
+    # Hide TransposedConv1DLayer for old Theano versions
+    del TransposedConv1DLayer, Deconv1DLayer
+    __all__.remove('TransposedConv1DLayer')
+    __all__.remove('Deconv1DLayer')
+    
+if not hasattr(T.nnet.abstract_conv,
+               'AbstractConv2d_gradInputs'):  # pragma: no cover
+    # Hide TransposedConv2DLayer for old Theano versions
+    del TransposedConv2DLayer, Deconv2DLayer
+    __all__.remove('TransposedConv2DLayer')
+    __all__.remove('Deconv2DLayer')
```

Comment:
The 2D version is always available in the Theano versions we support. I'd prefer to only add the check for 1D, directly after the 1D definitions.